compiler/src/dmd/backend/arm/cod2.d

```diff
@@ -321,6 +321,192 @@ void cddiv(ref CGstate cg, ref CodeBuilder cdb,elem* e,ref regm_t pretregs)
     fixresult(cdb,e,retregs,pretregs);
 }
 
+/***************************
+ * Handle OPnot and OPbool.
+ * Generate:
+ *      c:      [evaluate e1]
+ *      cfalse: [save reg code]
+ *              clr     reg
+ *              jmp     cnop
+ *      ctrue:  [save reg code]
+ *              clr     reg
+ *              inc     reg
+ *      cnop:   nop
+ */
+
+@trusted
+void cdnot(ref CGstate cg, ref CodeBuilder cdb,elem* e,ref regm_t pretregs)
+{
+    //printf("cdnot()\n");
+    reg_t reg;
+    regm_t forflags;
+    regm_t retregs;
+    elem* e1 = e.E1;
+
+    if (pretregs == 0)
+        goto L1;
+    if (pretregs == mPSW)
+    {   //assert(e.Eoper != OPnot && e.Eoper != OPbool);*/ /* should've been optimized
+    L1:
+        codelem(cgstate,cdb,e1,pretregs,false);      // evaluate e1 for cc
+        return;
+    }
+
+    OPER op = e.Eoper;
+    uint sz = tysize(e1.Ety);
+    uint rex = (I64 && sz == 8) ? REX_W : 0;
+    uint grex = rex << 16;
+
+    if (0 && !tyfloating(e1.Ety))
```

Comment:
why have an trivially false branch?

Reply:
Tomorrow I will investigate if it's worth while to use that branch.

Reply:
redone